lib/cocoapods/generator/prepare_artifacts_script.rb

```diff
@@ -0,0 +1,248 @@
+            local target_variant=""
+            if [[ "$PLATFORM_NAME" == *"simulator" ]]; then
+              target_variant="simulator"
+            fi
+            if [[ "$EFFECTIVE_PLATFORM_NAME" == *"maccatalyst" ]]; then
+              target_variant="maccatalyst"
+            fi
+            for i in ${!paths[@]}; do
+              if [[ "${paths[$i]}" == *"$target_arch"* ]] && [[ "${paths[$i]}" == *"$target_variant"* ]]; then
+                # Found a matching slice
+                echo "Selected xcframework slice ${paths[$i]}"
+                target_path=${paths[$i]}
+                break;
+              fi
+            done
+            
+            if [[ -z "$target_path" ]]; then
+              echo "warning: [CP] Unable to find matching .xcframework slice in '${paths[@]}' for the current build architectures ($ARCHS)."
+              return
+            fi
+
+            install_framework "$basepath/$target_path" "$embed"
+          }
+
+        SH
+        contents_by_config = Hash.new do |hash, key|
+          hash[key] = ""
+        end
+        xcframeworks_by_config.each do |config, xcframeworks|
+          next if xcframeworks.empty?
+          xcframeworks.each do |xcframework|
+            dynamic_slices, static_slices = xcframework.slices
+                                              .select { |f| f.platform.symbolic_name == platform.symbolic_name }
+                                              .partition { |slice| Xcode::LinkageAnalyzer.dynamic_binary?(slice.binary_path) }
+            next if dynamic_slices.empty? && static_slices.empty?
+            unless dynamic_slices.empty?
+              args = install_xcframework_args(xcframework.path, dynamic_slices, false)
+              contents_by_config[config] << %(  install_xcframework #{args}\n)
+            end
+
+            unless static_slices.empty?
+              args = install_xcframework_args(xcframework.path, static_slices, true)
+              contents_by_config[config] << %(  install_xcframework #{args}\n)
+            end
+
+            dsyms = PrepareArtifactsScript.dsym_paths(xcframework.path)
+            dsyms.each do |path|
+              source = shell_escape("${PODS_ROOT}/#{path.relative_path_from(sandbox_root).to_s}")
+              contents_by_config[config] << %(  install_artifact #{source} "${TARGET_BUILD_DIR}" "true"\n)
+            end
+          end
+        end
+
+        script << "\n" unless contents_by_config.empty?
+        contents_by_config.keys.sort.each do |config|
+          contents = contents_by_config[config]
+          next if contents.empty?
+          script << %(if [[ "$CONFIGURATION" == "#{config}" ]]; then\n)
+          script << contents
+          script << "fi\n"
+        end
+
+        script << "\necho \"Artifact list stored at $ARTIFACT_LIST_FILE\"\n"
+        script << "\ncat \"$ARTIFACT_LIST_FILE\"\n"
+        script
+      end
+
+      def shell_escape(value)
+        "\"#{value}\""
+      end
+
+      private
```

Comment:
is this needed?

Reply:
would you prefer the function below here be public instead?

Reply:
nope if it doesnt need to be.